core/src/main/scala/spinal/core/AFix.scala

```diff
@@ -649,17 +692,21 @@ class AFix(val maxValue: BigInt, val minValue: BigInt, val exp: Int) extends Mul
   def floor(exp : Int): AFix = {
     val drop = exp-this.exp
     if(drop < 0) return CombInit(this)
-    val res = new AFix(this.maxValue >> drop, this.minValue >> drop, 0)
+    val res = new AFix(this.maxRaw >> drop, this.minRaw >> drop, 0)
```

Comment:
Should `res` be 
```
val res = new AFix(this.maxValue >> drop, this.minValue >> drop, exp)
```
?

Reply:
That's what I get for copy and pasting. Will fix

Reply:
Thanks! LGTM :)